tests/serverpod_test_server/test/generated_methods/from_json_test.dart

```diff
@@ -0,0 +1,303 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test(
+      'Given a non-nullable field, when deserializing from JSON with null value, then a TypeError is thrown',
+      () {
+    expect(
+      () => SimpleData.fromJson({"num": null}),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'Given a class with non-nullable fields, when deserializing from an empty JSON, then a TypeError is thrown',
+      () {
+    expect(
+      () => SimpleData.fromJson({}),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'Given a class with a nullable integer field, when deserializing from JSON with correct, incorrect, and null values, then proper behavior is observed',
```

Comment:
Suggestion: Split into separate tests (and use a group to denote the "Given a class with a nullable integer field," part)

There are two important reasons to split them up, the first is that if the test fails, say in our CI, it is not clear what is broken from the three scenarios. The second reason is that we won't run any checks after the first expect fails.

Reply:
But this is a really good grouping! This makes it really clear what scenarios are covered for an int field.